### PR TITLE
DROOLS-3546 DMN fix FEEL context entry when key is a string literal

### PR DESCRIPTION
**Pull request by tarilabs**

### CODE REVIEW

File: kie-dmn/kie-dmn-feel/src/main/java/org/kie/dmn/feel/parser/feel11/ASTBuilderVisitor.java [hunk 1]
```diff
@@ -285,7 +285,12 @@ public BaseNode visitIterationNameDefinition(FEEL_1_1Parser.IterationNameDefinit
 
     @Override
     public BaseNode visitKeyString(FEEL_1_1Parser.KeyStringContext ctx) {
-        return ASTBuilderFactory.newNameDefNode( ctx, ctx.getText() );
```
Comment (tarilabs) [inline kie-dmn/kie-dmn-feel/src/main/java/org/kie/dmn/feel/parser/feel11/ASTBuilderVisitor.java:288]: I believe this line in the previous version was the original bug.
This PR raise a proposal for a potential way to fix it.

**Review by tarilabs — COMMENTED** [review]
/cc @etirelli @baldimir @jiripetrlik

**Comment by tarilabs** [comment]
(edited comment) please note I cannot "just" substitute the String (literal) with a NameDef during parse time as anyway `ParserHelper` is relying on the ANLTR scope (immutables) to determine what name to push or variable to define. I stand by the proposal as raised in the PR code.

**Review by etirelli — APPROVED** [review]

**Comment by etirelli** [comment]
I am approving this PR because I see in the FEEL grammar it supports string literals as keys, but I think this is a mistake in the spec.

**Pull request merged**